nntrainer/graph/network_graph.cpp

```diff
@@ -86,7 +86,9 @@ void NetworkGraph::addLayerNode(std::shared_ptr<LayerV1> layer) {
 
 void NetworkGraph::countNonTrainableLayersAtBegin() {
   for (auto iter = cbegin(); iter != cend(); iter++) {
-    if ((*iter)->getObject()->getTrainable()) {
+    // TODO: check if getTrainable() was set and if trainable weights exist,
+    // then throw
+    if ((*iter)->getTrainable() && (*iter)->supportBackwarding()) {
```

Comment:
Could you elaborate the difference between getTrainable and supportBackwarding?

Reply:
getTrainable() means if the layer should be trained (user can set trainable = true/false for a layer).
supportBackwarding() means if the layer provides implementation to do backwarding (eg. input_layer does not support backwarding as calcDerivative is not implemented).

Reply:
So.. for making clearer, for the Conv2D layer, supportBackwarding() is always true ( it looks more like property then ), and can be trainable true or false depending on option then. we also have training variable and weight also have trainable property. So if the layer trainable is set true, then all the trainable in weights for that layer is going to set trainable.. ?